elf_reader.go

```diff
@@ -413,44 +426,69 @@ func (ec *elfCode) relocateInstruction(ins *asm.Instruction, rel elf.Symbol) err
 		}
 
 	case programSection:
-		if ins.OpCode.JumpOp() != asm.Call {
-			return fmt.Errorf("not a call instruction: %v", ins)
+		if ins.OpCode.JumpOp() != asm.Call && !ins.OpCode.IsDWordLoad() {
+			return fmt.Errorf("neither a call nor a load instruction: %v", ins)
 		}
 
-		if ins.Src != asm.PseudoCall {
+		if ins.OpCode.JumpOp() == asm.Call && ins.Src != asm.PseudoCall {
 			return fmt.Errorf("call: %s: incorrect source register", name)
 		}
 
 		switch typ {
```

Comment:
I think it might be better to switch on ins.OpCode and then deal with `typ` instead of the other way round. That makes introducing unwanted behaviour for a given opcode less likely.